### PR TITLE
Make sure we can run the same python script n-times.

### PR DESCRIPTION
**Pull request by gz**
Fixes logic that check if an entity already exists for program and pipeline.

### CODE REVIEW

File: python/dbsp/connection.py [hunk 1]
```diff
@@ -58,20 +66,36 @@ def create_or_replace_program(self, *, name: str, sql_code: str, description: st
...
-            program_id = resp.unwrap("Failed to unwrap program %s" % (name))[0].program_id
-            resp = delete_program.sync_detailed(client = self.api_client, program_id = program_id)
-
-        # Create a new one instead
-        request = NewProgramRequest(name=name, code=sql_code, description='')
-        new_program_response = new_program.sync_detailed(client = self.api_client, json_body=request).unwrap("Failed to create a program")
+            program_id = resp.unwrap("Failed to unwrap program %s" % (name))[
+                0
+            ].program_id
+            # Create a new one instead
```
Comment (ryzhyk) [inline python/dbsp/connection.py:80]: wrong comment

File: python/dbsp/connection.py [hunk 2]
```diff
@@ -58,20 +66,36 @@ def create_or_replace_program(self, *, name: str, sql_code: str, description: st
             DBSPProgram
         """
 
-        return self.create_program_inner(name = name, sql_code = sql_code, description = description, replace = True)
+        return self.create_program_inner(
+            name=name, sql_code=sql_code, description=description
+        )
 
-    def create_program_inner(self, *, name: str, sql_code: str, description: str, replace: bool):
+    def create_program_inner(self, *, name: str, sql_code: str, description: str):
```
Comment (ryzhyk) [inline python/dbsp/connection.py:73]: Seems wrong to always replace existing program.  The `replace` flag may have been useful.  But this whole API will probably get refactored, so I'm ok with this.

File: python/dbsp/connection.py [hunk 3]
```diff
@@ -58,20 +66,36 @@ def create_or_replace_program(self, *, name: str, sql_code: str, description: st
...
+            program_id = resp.unwrap("Failed to unwrap program %s" % (name))[
+                0
+            ].program_id
+            # Create a new one instead
+            request = UpdateProgramRequest(
+                name=name, code=sql_code, description=description
+            )
+            program_response = update_program.sync_detailed(
+                client=self.api_client, program_id=program_id, json_body=request
+            ).unwrap("Failed to create a program")
```
Comment (ryzhyk) [inline python/dbsp/connection.py:86]:
```suggestion
            ).unwrap("Failed to update the program")
```

**Review by ryzhyk — APPROVED** [review]

File: python/dbsp/connection.py (hunk 2 above)
Comment (gz) [reply python/dbsp/connection.py:73]: in pipeline and connector we also update if something with this name already exists otherwise its created

**Pull request merged**